squidpy/pl/_var_by_distance.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+from pathlib import Path
+from types import MappingProxyType
+from typing import Any, List, Mapping, Optional, Sequence, Tuple, Union
+
+import matplotlib.cm as cm
+import matplotlib.colors as colors
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+from anndata import AnnData
+from cycler import Cycler
+from matplotlib import rcParams
+from matplotlib.axes import Axes
+from scanpy.plotting._tools.scatterplots import _panel_grid
+from scanpy.plotting._utils import _set_default_colors_for_categorical_obs
+from scipy.sparse import issparse
+
+from squidpy.pl._utils import save_fig
+
+__all__ = ["var_by_distance"]
+
+
+def var_by_distance(
+    adata: AnnData,
+    design_matrix_key: str,
+    var: str | List[str],
+    anchor_key: str | List[str],
+    color: str | None = None,
+    covariate: str | None = None,
+    order: int = 5,
+    show_scatter: bool = True,
+    dpi: int | None = None,
+    figsize: Tuple[int, int] | None = None,
+    line_palette: Union[str, Sequence[str], Cycler, None] = None,
+    scatter_palette: Union[str, Sequence[str], Cycler, None] = "viridis",
+    save: str | Path | None = None,
+    return_ax: Optional[bool] = None,
+    regplot_kwargs: Mapping[str, Any] = MappingProxyType({}),
+    scatterplot_kwargs: Mapping[str, Any] = MappingProxyType({}),
+) -> Union[Axes, None]:
+    """
+    Plot var expression by distance to anchor point.
+
+    Parameters
+    ----------
+    adata
+        Annotated data matrix.
+    design_matrix_key
+        Name of the design matrix, previously computed with :func:`squidpy.tl.var_by_distance`, to use.
+    var
+        Variables to plot on y-axis.
+    anchor_key
+        anchor point column from which distances are taken.
+    color
+        Variables to plot on color palette.
+    covariate
+        A covariate for which separate regression lines are plotted for each category.
+    order
+        Order of the polynomial fit for :func:`seaborn.regplot`.
+    dpi
+        Dpi value of the plot.
+    figsize
+        Size of the plot in inches.
+    line_palette
+        Categorical color palette used in case a covariate is specified.
+    scatter_palette
+        Color palette for the scatter plot underlying the `sns.regplot`
+    save
+        Path to save the figure to.
+    return_ax
+        Whether to return `matplotlib.axes.Axes` object(s).
+    regplot_kwargs
+        kwargs for `sns.regplot`
+    scatterplot_kwargs
+        kwargs for `sns.scatter`
+    %(plotting_save)s
+    %(plotting_dpi)s
+    %(plotting_figsize)s
```

Comment:
docstrings are still wrong:
- typos e.g. "anchor point" instead of "Anchor point", if decide to capitalize, always capitalize, if decide to not capitalize, always don't capitalize
- the docstrings processor variable like this `%(plotting_save)s` already have some of those arguments spelled out. E.g. I'm sure that save is somewhere in `%(plotting_save)s` yet you spell out save also above. The reason for docrep variables is to have consutency of docstrings for arguments that are found often in the code. So for all the variables that are present in other plotting functions (e.g. save, but also dpi and others) search if there is a variable that already contains them and if so, use that. The docrep variables are store din `_docs.py` file.